tools/condition.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2022, seL4 Project a Series of LF Projects, LLC
+#
+# SPDX-License-Identifier: BSD-2-Clause or GPL-2.0-only
+#
+
+def remove_ws_comments(node):
+    for child in list(node.childNodes):
+        if child.nodeType == node.COMMENT_NODE or (child.nodeType == node.TEXT_NODE and len(child.data.strip()) == 0):
+            node.removeChild(child)
+
+
+def condition_to_cpp(conditions):
+    n = len(conditions)
+    assert n <= 1
+    if n == 0:
+        return ""
+    else:
+        remove_ws_comments(conditions[0])
+        children = conditions[0].childNodes
+        if children == None or len(children) == 0:
```

Comment:
Do we really have to check for `None` explicitly?
```suggestion
        if not children or len(children) == 0:
```